File: apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/NativeModulesPackage.java
package com.ledger.live;


import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class NativeModulesPackage implements ReactPackage {

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }

    // Returns a list of native modules to be registered by React Native.
    @Override
    public List<NativeModule> createNativeModules(
            ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new BluetoothHelperModule(reactContext));
        modules.add(new ImagePickerModule(reactContext));
        modules.add(new LocationHelperModule(reactContext));
        return modules;
    }

}

File: apps/ledger-live-mobile/android/app/src/main/java/com/ledger/live/NativeModulesPackage.kt
package com.ledger.live

import android.view.View
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ReactShadowNode
import com.facebook.react.uimanager.ViewManager
import com.ledger.live.fabric.MeasureTransformModule
import kotlinx.coroutines.Dispatchers

class NativeModulesPackage : ReactPackage {

    override fun createViewManagers(
        reactContext: ReactApplicationContext
    ): MutableList<ViewManager<View, ReactShadowNode<*>>> = mutableListOf()

    override fun createNativeModules(
        reactContext: ReactApplicationContext
    ): MutableList<NativeModule> = buildList {
        add(BluetoothHelperModule(reactContext))
        add(LocationHelperModule(reactContext, coroutineDispatcher = Dispatchers.Default))
        add(MeasureTransformModule(reactContext))

        // Registered unconditionally, and deliberately so: the previous
        // availability probe called System.loadLibrary during bridge
        // initialisation on every launch, whether or not anything Zcash was
        // used. Instantiating the module is free; the library loads on the
        // first actual call, and a build without the engine rejects with
        // ZCASH_FFI_UNAVAILABLE, which the JS layer already handles.
        add(ZcashFfiModule(reactContext))
    }.toMutableList()
}